### ISSUE TITLE
[BUG] TiSpark breaks CTE

### ISSUE THREAD

**Issue by Dieken** [pingcap/tispark#1624]
**Describe the bug**

Expected:
```
spark-2.4.6-bin-hadoop2.7/bin/spark-shell

scala> spark.sql("with t1 as (select 11) select * from t1").show
+---+
| 11|
+---+
| 11|
+---+
```

Wrong:
```
$ spark-2.4.6-bin-hadoop2.7/bin/spark-shell --conf spark.sql.extensions=org.apache.spark.sql.TiExtensions --conf spark.tispark.pd.addresses=127.0.0.1:2379 --jars tispark-assembly-2.3.1.jar

scala> spark.sql("with t1 as (select 11) select * from t1").show
org.apache.spark.sql.AnalysisException: Table or view not found: `default`.`t1`; line 1 pos 37;
'Project [*]
+- 'UnresolvedRelation `default`.`t1`

  at org.apache.spark.sql.catalyst.analysis.package$AnalysisErrorAt.failAnalysis(package.scala:42)
  at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$checkAnalysis$1.apply(CheckAnalysis.scala:91)
  at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$checkAnalysis$1.apply(CheckAnalysis.scala:86)
  at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:126)
  at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$foreachUp$1.apply(TreeNode.scala:125)
  at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$foreachUp$1.apply(TreeNode.scala:125)
  at scala.collection.immutable.List.foreach(List.scala:392)
  at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:125)
  at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$class.checkAnalysis(CheckAnalysis.scala:86)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.checkAnalysis(Analyzer.scala:95)
  at org.apache.spark.sql.catalyst.analysis.Analyzer$$anonfun$executeAndCheck$1.apply(Analyzer.scala:108)
  at org.apache.spark.sql.catalyst.analysis.Analyzer$$anonfun$executeAndCheck$1.apply(Analyzer.scala:105)
  at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:201)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:105)
  at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:58)
  at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:56)
  at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:48)
  at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:78)
  at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:643)
  ... 51 elided
```

**Spark and TiSpark version info**

* spark-2.4.6-bin-hadoop2.7
* TiSpark 2.3.1
* macOS 10.15.6
* openjdk version "1.8.0_262"
  OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_262-b10) 
  OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.262-b10, mixed mode)

**Comment by birdstorm** [comment]
@Dieken Hi, thanks for your report, can you specify the catalog of `t1`? It seems that you did not use `use $database` statement, so the default catalog will be `default`.

**Comment by Dieken** [comment]
@birdstorm  It's a table name for common table expression, not any real table in some database.  I tried `use default` and `...from t1` or `..from default.t1`,  all throw similar exception.

**Comment by birdstorm** [comment]
> @birdstorm  It's a table name for common table expression, not any real table in some database.  I tried `use default` and `...from t1` or `..from default.t1`,  all throw similar exception.

i see, is your table created on hive?

**Comment by birdstorm** [comment]
working on it.

**Comment by birdstorm** [comment]
it seems to be a bug since Spark v2.4.6, we will fix this along with support for [Spark v2.4.7].

**Comment by birdstorm** [comment]
In Spark v2.4.7, the problem no longer exists.
